go/vt/orchestrator/logic/tablet_discovery.go

```diff
@@ -285,21 +311,32 @@ func setReplicationSource(ctx context.Context, replica *topodatapb.Tablet, prima
 	return tmc.SetReplicationSource(ctx, replica, primary.Alias, 0, "", true, semiSync)
 }
 
-// shardPrimary finds the primary of the given keyspace-shard by reading the topo server
-func shardPrimary(ctx context.Context, keyspace string, shard string) (primary *topodatapb.Tablet, err error) {
-	si, err := ts.GetShard(ctx, keyspace, shard)
-	if err != nil {
-		return nil, err
-	}
-	if !si.HasPrimary() {
-		return nil, fmt.Errorf("no primary tablet for shard %v/%v", keyspace, shard)
-	}
-	// TODO(GuptaManan100): Instead of another topo call, use the local information by calling
-	// ReadTablet. Currently this isn't possible since we only have the primary alias and not the source host and port
-	// This should be fixed once the tablet alias is changed to be the primary key of the table
-	primaryInfo, err := ts.GetTablet(ctx, si.PrimaryAlias)
-	if err != nil {
-		return nil, err
+// shardPrimary finds the primary of the given keyspace-shard by reading the orchestrator backend
+func shardPrimary(keyspace string, shard string) (primary *topodatapb.Tablet, err error) {
+	query := `SELECT
+		info,
+		hostname,
+		port,
+		tablet_type,
+		primary_timestamp
+	FROM 
+		vitess_tablet
+	WHERE
+		keyspace = ? AND shard = ?
+		AND tablet_type = ?
+	ORDER BY
+		primary_timestamp DESC
+	LIMIT 1
+`
+	err = db.Db.QueryOrchestrator(query, sqlutils.Args(keyspace, shard, topodatapb.TabletType_PRIMARY), func(m sqlutils.RowMap) error {
```

Comment:
As part of cleaning up, we should rename some of these functions. `QueryOrchestrator` is not the right name for this function. Similarly we have `OpenOrchestrator` too.

Reply:
Yes, I had it in my mind to get rid of the `Orchestrator` references everywhere, from Parameters, flags, package names, function names, etc. I'll do in a follow-up PR so that it is easier to review